lua/orgmode/org/syntax.lua

```diff
@@ -36,7 +42,73 @@ local function add_todo_keywords_to_spellgood()
   end
 end
 
+local function _display_cookie_virt_text(cookie, checked, total)
+  local row, column = cookie:start()
+  local cookie_text = vim.treesitter.get_node_text(cookie, 0)
+
+  -- TODO: fix this in treesitter
+  local _, leading_spaces = cookie_text:find('^%s+')
+  cookie_text = vim.trim(cookie_text)
+
+  local text
+  if cookie_text == '[/]' then
+    text = string.format('[%d/%d]', checked, total)
+  else
+    text = string.format('[%d%%]', (100 * checked) / total)
+  end
+  vim.api.nvim_buf_set_extmark(0, cookie_namespace, row, column + (leading_spaces or 0), {
+    -- TODO: make highlight group configurable
+    virt_text = { { text, 'Error' } },
+    virt_text_pos = 'overlay',
+    hl_mode = 'combine',
+  })
+end
+
+local function _get_cookie_checked_and_total(parent)
+  local parent_type = parent:type()
+  local start_row, _, end_row, _ = parent:range()
+  local checked, total = 0, 0
+  for _, checkbox in checkbox_query:iter_captures(parent, 0, start_row, end_row + 1) do
+    local closest_parent = utils.get_closest_parent_of_type(checkbox, parent_type)
+    if closest_parent and closest_parent == parent then -- only count direct children
```

Comment:
it would be nicer if the query could do this directly, but I don't know if that is possible.

Reply:
it doesn't need recursion. It should only check the first level of checkboxes. The treesitter query finds all nested ones as well, so I had to do this hacky check.

The implementation with just `iter_children` isn't super pretty either, but you are probably right, it's easier to understand and faster

```lua
local function _get_cookie_checked_and_total(parent)
  local checked, total = 0, 0
  for child in parent:iter_children() do
    if child:type() == 'listitem' then
      for listitem_child in child:iter_children() do
        if listitem_child:type() == 'checkbox' then
          local checkbox_text = vim.treesitter.get_node_text(listitem_child, 0)
          if checkbox_text:match('%[[x|X]%]') then
            checked = checked + 1
          end
          total = total + 1
        end
      end
    end
  end
  return checked, total
end
```